source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,143 @@
+your client for multiple sessions and transactions instead of
+instantiating a new client each time.
+
+.. warning::
+
+   Use a ``ClientSession`` only with the ``MongoClient`` (or associated
+   ``MongoDatabase`` or ``MongoCollection``) that created it. Using a
+   ``ClientSession`` with a different ``MongoClient`` results in operation
+   errors.
+
+Methods
+-------
+
+Create a ``ClientSession`` by using the ``startSession()`` method on your 
+``MongoClient``instance. You can then modify the session state by using the 
+methods provided by the ``ClientSession``. The following table details the 
+methods you can use to manage your transaction:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``startTransaction()``
+     - | Starts a new transaction for this session with the
+         default transaction options. Use the ``TransactionOptions`` 
+         parameter to start a transaction with given options. You 
+         cannot start a transaction if there's already an active transaction
+         on the session.
+       |
+       | **Parameter**: ``transactionOptions``
+
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. Returns an error 
+         if there is no active transaction for the
+         session or the transaction was previously ended.
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or if the
+         transaction was ended.
+
+   * - ``withTransaction()``
+     - | Starts a new transaction for this session and runs the given function.
+       |
+       | **Parameter**: ``transactionBody``
+
+A ``ClientSession`` also has methods to retrieve session properties and modify mutable 
+session properties. View the `API documentation <{+api+}/apidocs/mongodb-driver-sync/com/mongodb/client/ClientSession.html>`__ 
+to learn more about these methods.
+
+Example
+-------
+
+The following example demonstrates how you can create a session, create a transaction, 
+and commit a multi-document insert operation through the following steps:
+
+1. Create a session from the client using the ``startSession()`` method.
+#. Set transaction options to configure transactional behavior.
+#. Use the ``withTransaction()`` method to start a transaction.
+#. Insert multiple documents. The ``withTransaction()`` method executes, commits, and aborts 
+   the transaction. If any operation results in errors, ``withTransaction()`` handles canceling
+   the transaction.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/Transaction.java
+   :language: java
+   :start-after: start transaction
+   :end-before: end transaction
+
+If you require more control over your transactions, you can use the ``startTransaction()``
+method. This method allows you to explicitly commit or abort the transaction and manually
+manage the transaction lifecycle.
```

Comment:
S: if you use the startTxn() method, you also need to mention that the user will need to use the other methods to commit/handle errors